### ISSUE TITLE
EOS build failed on macOS 10.13

### ISSUE THREAD

**Issue by chencore** [EOSIO/eos#1000]
I run "./build.sh darwin"，but failed：
```
 44%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/sha1.cpp.o
In file included from /Users/chenke/eos/eos/libraries/fc/src/log/gelf_appender.cpp:5:
In file included from /Users/chenke/eos/eos/libraries/fc/include/fc/log/gelf_appender.hpp:3:
/Users/chenke/eos/eos/libraries/fc/include/fc/log/appender.hpp:6:42: error: 
      definition of type 'io_service' conflicts with typedef of the same name
namespace boost { namespace asio { class io_service; } }
                                         ^
/usr/local/include/boost/asio/io_service.hpp:27:20: note: 'io_service' declared
      here
typedef io_context io_service;
                   ^
In file included from /Users/chenke/eos/eos/libraries/fc/src/log/gelf_appender.cpp:5:
/Users/chenke/eos/eos/libraries/fc/include/fc/log/gelf_appender.hpp:7:42: error: 
      definition of type 'io_service' conflicts with typedef of the same name
namespace boost { namespace asio { class io_service; } }
                                         ^
/usr/local/include/boost/asio/io_service.hpp:27:20: note: 'io_service' declared
      here
typedef io_context io_service;
                   ^
In file included from /Users/chenke/eos/eos/libraries/fc/src/log/gelf_appender.cpp:5:
/Users/chenke/eos/eos/libraries/fc/include/fc/log/gelf_appender.hpp:31:10: warning: 
      'initialize' overrides a member function but is not marked 'override'
      [-Winconsistent-missing-override]
    void initialize(boost::asio::io_service& io_service);
         ^
/Users/chenke/eos/eos/libraries/fc/include/fc/log/appender.hpp:44:23: note: 
      overridden virtual function is here
         virtual void initialize( boost::asio::io_service& io_service ) = 0;
                      ^
1 warning and 2 errors generated.
make[2]: *** [libraries/fc/CMakeFiles/fc.dir/src/log/gelf_appender.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [libraries/fc/CMakeFiles/fc.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by ytiany** [comment]
same problem

**Comment by kmanikumar** [comment]
same as #984

**Comment by yeliang** [comment]
not resolved yet ... waiting for ...

**Comment by randytorres** [comment]
Any updates on this?

**Comment by heifner** [comment]
Fixed by #1012
